### PR TITLE
composite-checkout: Implement Sofort payment method in composite checkout package and use it in calypso

### PR DESCRIPTION
**Pull request by nbloomf**
#### Changes proposed in this Pull Request

* Implement Sofort payment method in composite checkout.

Screen Shot 2020-06-25 at 5 17 16 PM

#### Testing instructions

* Apply D45520-code to force this payment method to be available (or use a vpn to spoof a German IP address), and enter checkout with Euros as your currency.

### CODE REVIEW

File: packages/composite-checkout/src/lib/payment-methods/sofort.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+			</PaymentMethodLogos>
+		</React.Fragment>
+	);
+}
+
+const SofortLogoUI = styled( SofortLogo )`
+	width: 28px;
+`;
+
+function SofortLogo() {
```
Comment (nbloomf) [inline packages/composite-checkout/src/lib/payment-methods/sofort.js:231]: Need a `className` parameter here

File: packages/composite-checkout/src/lib/payment-methods/sofort.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+	:after {
+		display: block;
+		width: calc( 100% - 6px );
+		height: 1px;
+		content: '';
+		background: ${ ( props ) => props.theme.colors.borderColorLight };
+		position: absolute;
+		top: 0;
+		left: 3px;
```
Comment (nbloomf) [inline packages/composite-checkout/src/lib/payment-methods/sofort.js:117]: Need to handle `.rtl` here... maybe handling that manually isn't a good idea

Comment (sirbrillig) [reply packages/composite-checkout/src/lib/payment-methods/sofort.js:117]: 😬  We'll probably need to handle that for the other payment methods too. Maybe worth doing all at once?

File: client/my-sites/checkout/composite-checkout/composite-checkout.js [hunk 3]
```diff
@@ -478,7 +478,9 @@ export default function CompositeCheckout( {
 			giropay: ( transactionData ) =>
 				genericRedirectProcessor( 'giropay', transactionData, getThankYouUrl, isWhiteGloveOffer ),
 			ideal: ( transactionData ) =>
-				genericRedirectProcessor( 'ideal', transactionData, getThankYouUrl, isWhiteGloveOffer ),
+                genericRedirectProcessor( 'ideal', transactionData, getThankYouUrl, isWhiteGloveOffer ),
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/composite-checkout.js:481]: Something seems off about this indentation 🤔

File: client/my-sites/checkout/composite-checkout/payment-method-helpers.js [hunk 4]
```diff
@@ -125,6 +125,16 @@ export async function submitStripeRedirectTransaction( paymentMethodId, transact
 	return submit( formattedTransactionData );
 }
 
+export async function submitStripeSofortTransaction( transactionData, submit ) {
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/payment-method-helpers.js:128]: You should be able to get rid of this now that we have the generic one (I think this isn't being used).

File: client/my-sites/checkout/composite-checkout/use-create-payment-methods.js [hunk 5]
```diff
@@ -284,6 +319,7 @@ export default function useCreatePaymentMethods( {
 		...existingCardMethods,
 		idealMethod,
 		giropayMethod,
+        sofortMethod,
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/use-create-payment-methods.js:322]: Possssibly also an indent issue here?

**Pull request merged**